src/components/submission/table.vue

```diff
@@ -90,34 +120,36 @@ export default {
       return this.project != null && this.project.permits('submission.update');
     }
   },
-  mounted() {
-    let index;
-    let metadataRow;
-    const { metadataBody, dataBody } = this.$refs;
-    const mouseover = (event) => {
-      const { dataset } = event.target.closest('tr');
-      if (dataset.index !== index) {
-        if (index != null) metadataRow.classList.remove('actions-shown');
-        index = dataset.index;
-        metadataRow = metadataBody.querySelector(`tr:nth-child(${index})`);
-        // The SubmissionMetadataRow element does not have a class binding, so I
-        // think we can add this class without Vue removing it.
-        metadataRow.classList.add('actions-shown');
-      }
-    };
-    const mouseleave = () => {
-      if (index != null) {
-        metadataRow.classList.remove('actions-shown');
-        index = null;
-        metadataRow = null;
+  methods: {
+    setActionsTrigger(trigger) {
+      this.actions.trigger = trigger;
+    },
+    toggleHoverClass(event) {
+      const dataRow = event.target.closest('tr');
+      const index = Number.parseInt(dataRow.dataset.index, 10);
+      if (index === this.actions.dataHover) return;
+      const { metadataBody } = this.$refs;
+      if (this.actions.dataHover != null)
+        metadataBody.querySelector('.data-hover').classList.remove('data-hover');
+      const metadataRow = metadataBody.querySelector(`tr:nth-child(${index + 1})`);
+      // The SubmissionMetadataRow element does not have a class binding, so I
+      // think we can add this class without Vue removing it.
+      metadataRow.classList.add('data-hover');
```

Comment:
I've renamed this class from `actions-shown` to `data-hover`, since now hovering over a `SubmissionDataRow` doesn't necessarily mean that actions are shown for the row (because the user might be using the keyboard to navigate). The only purpose of `data-hover` is to indicate whether the cursor is over the `SubmissionDataRow`.